src/App.js

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import "./App.css";
+import { getSubCategory } from './helpers/getSubCategory';
+import {HOMEPAGE_CONSTANTS}  from "./constants.general";
+import { Route, Routes } from "react-router";
+import Homepage from './pages/homepage/Homepage';
+import Checkout from './pages/checkout/Checkout';
+
+
+class App extends React.Component {
+  
+  constructor(props) {
+    super(props);
+    this.state = {
+      category: HOMEPAGE_CONSTANTS.DEFAULT_CATEGORY,
+      currSubCategory:HOMEPAGE_CONSTANTS.DEFAULT_SUB_CATEGORY,
+      cartCount: 0,
+      cartOriginal: 0,
+      cartDiscount: 0
+    };
+  }
+  categoryClick=(event)=>{
+    this.setState({
+      category: event.target.innerText,
+      currSubCategory: "All"  
+    });
+
+  }
+  subCategoryOnClick=(event)=>{
+    this.setState({
+        currSubCategory: event.target.innerText
+      })
+  }
+  addProduct=(parentNode)=>{
+    let productId=parentNode.parentNode.getElementsByClassName("product-id")[0].innerText;
```

Comment:
Since we are just adding and removing the item from local storage, we don't need to pass this from parent component